### ISSUE TITLE
Search corrects spelling without telling the user

### ISSUE THREAD

**Issue by astrojuanlu** [readthedocs/addons#84]
I searched for "context" but started getting results for "content", which was confusing.

![Screenshot 2021-09-28 at 13-47-58 Search — Read the Docs 5 24 0 documentation](https://user-images.githubusercontent.com/316517/135081107-b5c845b7-a722-4e5a-b0fa-bb7b2abcb045.png)

**Comment by stsewd** [comment]
yeah, we can control that at search time, it's under a feature flag, since it can be slow

**Comment by humitos** [comment]
OK. Then, the work required here would be:

- [ ] Add an option in the Search Addons to show a `Fuzzy` checkbox in the UI
- [ ] Send a `?fuzzy=true|false` attribute to the endpoint
- [ ] Update the Search API endpoint to receive the `?fuzzy` attribute
- [ ] Modify backend to override the feature flag with the value from `?fuzzy` attribute

Anything else you have in mind?

**Comment by stsewd** [comment]
I think we should still have it under a feature flag until we prioritize optimizing ES.

**Comment by humitos** [comment]
@stsewd well, the goal of this issue is to remove the feature flag and expose the feature to users.

What does it mean it can be slow? Will that kill our servers? Can we add a timeout to ES queries as we do with PostgreSQL?

**Comment by humitos** [comment]
It seems we can specify a timeout on each query, https://www.elastic.co/guide/en/elasticsearch/reference/current/search-your-data.html#search-timeout, but also there is a `search.default_search_timeout` that we can set in the cluster. We should use that option.

**Comment by humitos** [comment]
> It seems we can specify a timeout on each query, [elastic.co/guide/en/elasticsearch/reference/current/search-your-data.html#search-timeout](https://www.elastic.co/guide/en/elasticsearch/reference/current/search-your-data.html#search-timeout), but also there is a `search.default_search_timeout` that we can set in the cluster

This would be also useful for the case described in https://github.com/readthedocs/readthedocs.org/issues/10321 👍🏼

**Comment by humitos** [comment]
@stsewd where does live the configuration of Elastic for production? How I can define the `search.default_search_timeout`? Do we have this in terraform?

I have the same questions for our development instance. Reading the docs, I didn't get _where_ should I define these settings. Do you know how to do this?
